### DISCUSSION TITLE
Week 4 of coding

### DISCUSSION THREAD

**Discussion by FlorianHeck2** [FlorianHeck2/codespaces-jupyter#4] [Announcements]
Hello together,
This week, i kept working on some bugfixes on my game, sadly without too mouch of progress, so i am going ti present you some Work i did on a nested for-loop:
I created some code producing this beautifull butterfly (don´t judge a book by it's cover :P )
<img width="170" alt="image" src="https://github.com/FlorianHeck2/codespaces-jupyter/assets/152873399/26001516-7761-4ea0-8a7b-6b2e101a1f42">
The code:

```py
max = 7 #max*2+1 results in maximal size
for x in range(2): #create sensor
    for l in range (max-1):
        print(" ", end=" ")
    print("0", end = " ")
    for k in range(int(max/5)):
        print(" ", end = " ")
    print("0", end = " ")
    print()

for x in range(2): #create head
    for l in range (max-1):
        print(" ", end=" ")
    for k in range(int(max/5)+2):
        print("0", end = " ")
    print()
    
for i in range(max): #Create first half of body and wing
    for j in range(i):
        print("0", end = " ")
    for m in range(max-i-1):
        print(" ", end = " ")
    for k in range(int(max/5)+2):
        print("0", end = " ")
    for n in range(max-i-1):
         print(" ", end = " ")
    for l in range(i):
         print("0", end=" ")
    print()

for i in range(max): # create second half and of body and wing
    for j in range(max-i-1):
        print("0", end = " ")
    for m in range(i):
        print(" ", end = " ")
    for k in range(int(max/5)+2):
        print("0", end = " ")
    for n in range(i):
         print(" ", end = " ")
    for l in range(max-i-1):
         print("0", end=" ")
    print()

for x in range(2): #create a tail
    for l in range (max-1):
        print(" ", end=" ")
    for k in range(int(max/5)+2):
        print("0", end = " ")
    print()

```

**Comment by laraanastasia** [comment]
I would find it really interesting, if you could tell us about what bugs you are currently trying to fix! 
Cool butterfly!

**Reply by FlorianHeck2** [reply]
The main problem I'm still working on is, that the walls of my labyrinth are still not detected correctly. You can't walk through them like you could during the last weeks, but when pressing different buttons quickly after each other and standing at a wall, you get able to cross it. I'll post you the Codepiece according to the problem later.

**Reply by FlorianHeck2** [reply]
Never mind, i found a way to make it work out, but if you want to have a look at it feel free (https://github.com/FlorianHeck2/codespaces-jupyter/blob/main/project/main.py)

**Comment by Rouge2123** [comment]
The butterfly looks really cool, I'm sure we can all appreciate the effort this took

**Reply by FlorianHeck2** [reply]
thanks

**Comment by NicoOblaender** [comment]
Pretty cool project. I like butterflies!

**Reply by FlorianHeck2** [reply]
thank you

**Comment by 1Wargear** [comment]
Looks good, is it possible to create the two halves with a function to avoid code repetition?

**Reply by FlorianHeck2** [reply]
haven't thought about that but I'll try it out!
